lib/gradebook.rb

```diff
@@ -0,0 +1,36 @@
+class Gradebook
+
+  attr_reader :instructor, :courses
+
+  def initialize(instructor)
+    @instructor = instructor
+    @courses = []
+  end
+
+  def add_course(course)
+    @courses << course
+  end
+
+  def list_all_students
+    all_enrolled_students = {}
+ 
+    @courses.each do |course|
+      all_enrolled_students[course] = course.students
+    end
+      
+  end
```

Comment:
This method does not return what you think needed it to return. And you likely assumed that this was doing what it was supposed to because there is no meaningful test that covers this functionality.